### PR TITLE
Replace pre-signed image by public url

### PR DESCRIPTION
**Pull request by Victor-van-Duijnen**
This PR replaces pre-signed image by public url. The image's path are hashed but publically accessible, limiting the number of calls to the application and its performance.

### CODE REVIEW

File: src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Helpers/MinIoLinkGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+﻿using HashidsNet;
+using Microsoft.Extensions.Options;
+using Smart.FA.Catalog.Infrastructure.Services.Options;
+
+namespace Smart.FA.Catalog.Infrastructure.Helpers;
+
+public class MinIoLinkGenerator : IMinIoLinkGenerator
```
Comment (LievenPraet) [inline src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Helpers/MinIoLinkGenerator.cs:7]: Service might be a better name to stay consistent in our naming
```suggestion
public class MinIoLinkService : IMinIoLinkService
```

File: src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Helpers/IMinIoLinkGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+﻿namespace Smart.FA.Catalog.Infrastructure.Helpers;
+
+public interface IMinIoLinkGenerator
```
Comment (LievenPraet) [inline src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Helpers/IMinIoLinkGenerator.cs:3]:
```suggestion
public interface IMinIoLinkService
```

File: src/UserAdmin/src/Smart.FA.Catalog.Web/Pages/Shared/_AdminLayout.cshtml [hunk 3]
```diff
@@ -53,8 +56,7 @@
                     <div class="c-toolbar__item">
                         <a class="c-user-navigation" href="#" data-menu="userMenu">
                             <div class="c-avatar c-avatar--img">
-                                <img src="/admin/myprofile?handler=LoadImage" alt="avatar">
-
+                                <img src="@(UserIdentity.CurrentTrainer.ProfileImagePath is null ? MinIoLinkGenerator.GetFullTrainerProfilePictureUrl(MinIoLinkGenerator.GetDefaultFullProfilePictureImageUrl()) : MinIoLinkGenerator.GetFullTrainerProfilePictureUrl(UserIdentity.CurrentTrainer.ProfileImagePath))" alt="avatar">
```
Comment (LievenPraet) [inline src/UserAdmin/src/Smart.FA.Catalog.Web/Pages/Shared/_AdminLayout.cshtml:59]: We could create a local function to make the code a bit more readable

**Review by LievenPraet — CHANGES_REQUESTED** [review]

File: src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Helpers/MinIoLinkGenerator.cs (hunk 1 above)
Comment (maxime-poulain) [reply src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Helpers/MinIoLinkGenerator.cs:7]: I suggested Victor that name, I think it is good as it makes it clear what it does.
An example:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.passwordhasher-1?view=aspnetcore-6.0

However MinIOService we have does a bunch of different things it makes sense have suffix.

You will also most likely name a service that provides IConnection IConnectionProvider.

File: src/UserAdmin/src/Smart.FA.Catalog.Web/Pages/Shared/_AdminLayout.cshtml (hunk 3 above)
Comment (maxime-poulain) [reply src/UserAdmin/src/Smart.FA.Catalog.Web/Pages/Shared/_AdminLayout.cshtml:59]: Two solutions here:
MinIoLinkGenerator.GetCurrentTrainerProfilePictureUrl() that uses the Current User Identity service to retrieves dynamically the URL with a fallback of the defualt one

OR

MinIoLinkGenerator.GetFullTrainerProfilePictureUrl(string path)=> if null return default;

But what's it is sure is that that line has too much logic.

File: src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Helpers/IMinIoLinkGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+{
+    /// <summary>
+    /// Create the relative URL of the profile picture of the trainer out of his id
+    /// </summary>
+    /// <param name="trainerId">The id of the trainer</param>
+    /// <returns>The relative url to the profile picture of the trainer</returns>
```
Comment (maxime-poulain) [inline src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Helpers/IMinIoLinkGenerator.cs:9]: It's more of a path nope?

File: src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Helpers/MinIoLinkGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        _storageOptions = storageOptions.Value;
+    }
+
+    /// <inheritdoc />
+    public string CreateTrainerProfilePictureUrl(int trainerId, string imageFormat)
+    {
+        // Generate unique salt value to avoid decoding
+        var hashIds = new Hashids(MinioLinkDefaultSaltValues.TrainerProfilePicture, 8);
+        return Path.Combine("trainers/", "profile-image/", $"{hashIds.Encode(trainerId)}{imageFormat}");
```
Comment (maxime-poulain) [inline src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Helpers/MinIoLinkGenerator.cs:21]: If you don't like putting that in the configuration files (fine by me) you should, i think, extract them in private constant at the top of the class

**Review by maxime-poulain — CHANGES_REQUESTED** [review]

**Review by maxime-poulain — APPROVED** [review]

**Review by LievenPraet — APPROVED** [review]

**Pull request merged**